### PR TITLE
ASoC: SOF: Intel: pci-mtl: fix ARL-S definitions, add ARL

### PR DESCRIPTION
**Pull request by plbossart**
The initial copy/paste from MTL was incorrect, the hardware is different and requires different descriptors along with a dedicated firmware binary.

Fixes: 3851831f529e ("ASoC: SOF: Intel: pci-mtl: use ARL specific firmware definitions")

### CODE REVIEW

**Review by ujfalusi — DISMISSED** [review]

**Review by aiChaoSONG — DISMISSED** [review]

File: sound/soc/sof/intel/pci-mtl.c [hunk 1]
```diff
@@ -50,6 +50,36 @@ static const struct sof_dev_desc mtl_desc = {
...
+		[SOF_IPC_TYPE_4] = "intel/sof-ipc4/arl",
+	},
+	.default_lib_path = {
+		[SOF_IPC_TYPE_4] = "intel/sof-ipc4-lib/arl",
+	},
+	.default_tplg_path = {
+		[SOF_IPC_TYPE_4] = "intel/sof-ace-tplg",
+	},
+	.default_fw_filename = {
+		[SOF_IPC_TYPE_4] = "sof-arl.ri",
```
Comment (kv2019i) [inline sound/soc/sof/intel/pci-mtl.c]: Wondering should we use "sof-arl-h.ri" to be consistent (and allow for different signing key). But probably this is good -- just need to use weirded cases for other variants if need arises.

**Review by kv2019i — DISMISSED** [review]
One inline comment, but overall looks good.

**Comment by plbossart** [comment]
V2: using "ARL" only in PCI ID and comments.

**Review by ujfalusi — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

File: sound/soc/sof/intel/pci-mtl.c [hunk 2]
```diff
@@ -80,10 +80,41 @@ static const struct sof_dev_desc arl_desc = {
...
+	.irqindex_host_ipc      = -1,
+	.chip_info = &arl_s_chip_info,
+	.ipc_supported_mask     = BIT(SOF_IPC_TYPE_4),
+	.ipc_default            = SOF_IPC_TYPE_4,
+	.dspless_mode_supported = true,         /* Only supported for HDaudio */
+	.default_fw_path = {
+		[SOF_IPC_TYPE_4] = "intel/sof-ipc4/arl",
+	},
+	.default_lib_path = {
+		[SOF_IPC_TYPE_4] = "intel/sof-ipc4-lib/arl",
```
Comment (ujfalusi) [inline sound/soc/sof/intel/pci-mtl.c:99]: @plbossart, @kv2019i, I have completely overlooked this, we should have the default paths:
```c
.default_fw_path = {
		[SOF_IPC_TYPE_4] = "intel/sof-ipc4/arl-s",
	},
	.default_lib_path = {
		[SOF_IPC_TYPE_4] = "intel/sof-ipc4-lib/arl-s",
	},
```
In order to follow the existing patterns. (not to reuse the ARL directory and use different firmware name).

Can we fix this up?

Comment (plbossart) [reply sound/soc/sof/intel/pci-mtl.c:99]: thanks @ujfalusi indeed, it's a miss. Care to send a fixup! patch?

**Pull request merged**